src/App.jsx

```diff
@@ -17,17 +28,51 @@ function App() {
     setUserList(prev => [...prev, user])
   }
 
+  const editUserList = (user) => {
+    setUserList( prev => {
+      const {name,age,index} = user;
+      const tempUserList = [...prev];
+      tempUserList[index] = {name,age};
+      return tempUserList;
+    }) 
+  }
+
+  const deleteFromUserList = (index) => {
+    setUserList(prev => {
+      const temp = [];
+      prev.forEach((elem, idx) => {
+        if(index !== idx ){
+          temp.push(elem);    
+        }
+      })
+      return temp;
+    })
+
+
+
+
+
+
+  }
+
+
+
   return (
     <main>
       
       {
         toggleComponents === "user-form" ? 
-        <UserForm handleToggleComponents={handleToggleComponents} 
-        addToUserList={addToUserList} /> :
-         toggleComponents === "user-table" ? 
-         <UserTable handleToggleComponents={handleToggleComponents}
-            userList={userList} /> : 
-            null
+          <UserForm handleToggleComponents={handleToggleComponents} 
+            addToUserList={addToUserList} /> :
+        toggleComponents === "user-table" ? 
+          <UserTable handleToggleComponents={handleToggleComponents}
+              userList={userList} setEditUserData={setEditUserData}
+              deleteFromUserList={deleteFromUserList}/> : 
+        toggleComponents === "user-edit-form" ? 
+              <UserEditForm editUserList={editUserList} 
+                setEditUserData={setEditUserData}
+                editUserData={editUserData}
+                handleToggleComponents={handleToggleComponents}/> : null
```

Comment:
use single component for creating and editing a user, don’t use separate components. Mange edit state through props